### DISCUSSION TITLE
Using sparse-checkout on Yarn Workspace

### DISCUSSION THREAD

**Discussion by BasixKOR** [yarnpkg/berry#5414] [General]
Hello, I've been exploring sparse-checkout for reducing disk usage and checkout speed on CI builds, but I've noticed Yarn breaks the lockfile in case not all of the workspace packages are checked-out.

I'd love to know if there's a way to use sparse-checkout with Yarn, or features to integrate it with Yarn.

**Comment by arcanis** [comment]
No, all workspace definitions need to be available for Yarn to work. The best you can do is use use `--filter` to avoid fetching the blobs from old commits.

**Reply by BasixKOR** [reply]
Can the workspace members be inferred from the lockfile? I *think* the lockfile has enough information to create the same dependency tree as if all workspace members are available, but I'm unsure about the details

**Reply by danielpza** [reply]
Perhaps it's possible with the plugin system?

**Comment by BasixKOR** [comment]
Potentially related: #1223
